### PR TITLE
feat(migrate): create built-in project for migrate and add hook to execute pre logic

### PR DESCRIPTION
**Pull request by yiminpeng**
#### What this PR does / why we need it:

1. create built-in project for migrator.
2. delete some yaml when ODC version less than 4.3.0.

#### Which issue(s) this PR fixes:
Fixes #

#### Additional documentation e.g., usage docs, etc.:

### CODE REVIEW

File: server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/Migrates.java [hunk 1]
```diff
@@ -85,7 +88,15 @@ public void migrate() {
                 version2Migratables.keySet().stream().map(Version::new).sorted().collect(Collectors.toList());
         log.info("versionCount={}", sortedVersions.size());
 
-        degradeCheck(sortedVersions.get(sortedVersions.size() - 1).getVersion());
+        String currentVersion = sortedVersions.get(sortedVersions.size() - 1).getVersion();
+
+        if (VersionUtils.isLessThan(configuration.getInitVersion(), "4.3.0.0")) {
```
Comment (MarkPotato777) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/Migrates.java:93]: This logic should be moved to the apsara implementation.

File: server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BeforeCheckDeleteHook.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+package com.oceanbase.odc.core.migrate;
+
+import javax.sql.DataSource;
+
+/**
+ * @author yiminpeng
+ * @date 2024/07/26
+ * @version : BeforeCheckDeleteHook.java, v 1.0
+ */
+public interface BeforeCheckDeleteHook {
```
Comment (MarkPotato777) [inline server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BeforeCheckDeleteHook.java:25]: It is unnecessary to limit it to `delete`. `MigratePreHook` is better?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/iam/auth/CustomAuthenticationSuccessHandler.java [hunk 3]
```diff
@@ -124,7 +125,12 @@ public void onAuthenticationSuccess(HttpServletRequest httpServletRequest, HttpS
             SecurityContextUtils.switchCurrentUserOrganization(user, team, httpServletRequest, true);
             // If bastion is enabled, every user must hold a built-in project for create temporary SQL console
             if (bastionEnabled) {
-                projectService.createProjectIfNotExists(user);
+                ProjectEntity project = projectService
+                        .createProjectIfNotExists(user,
+                                "USER_PROJECT_" + user.getAccountName(),
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/iam/auth/CustomAuthenticationSuccessHandler.java:130]: Is the original project name `'USER_PROJECT_xxx' + user.getAccountName()`?

File: server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BeforeCheckDeleteHook.java (hunk 2 above)
Comment (yiminpeng) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BeforeCheckDeleteHook.java:25]: Yes, considering expandability, deletion only needs to be reflected in the name of the method.

File: server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/Migrates.java (hunk 1 above)
Comment (yiminpeng) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/Migrates.java:93]: got it, done.

File: server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BeforeCheckDeleteHook.java (hunk 2 above)
Comment (yiminpeng) [reply server/odc-core/src/main/java/com/oceanbase/odc/core/migrate/BeforeCheckDeleteHook.java:25]: Yes, considering expandability, deletion only needs to be reflected in the name of the method.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/iam/auth/CustomAuthenticationSuccessHandler.java (hunk 3 above)
Comment (yiminpeng) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/iam/auth/CustomAuthenticationSuccessHandler.java:130]: Yes, `USER_PROJECT_` was as a constant.

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Pull request merged**